### PR TITLE
Video orientation change issue

### PR DESCRIPTION
**Pull request by jhaarpan**
**Issue #:**

**Description of changes:**
- Added an event listener to detect change of orientation and change the target canvas size when Background Blur/ Background Replacement is enabled.

**Testing:**
- Ran npm run build:release locally
- Tested the change manually by changing orientation on Android Chrome

*Can these tested using a demo application? Please provide reproducible step-by-step instructions.*
- Deploy the serverless demo of amazon-chime-sdk-js v2.25.0
- Start a meeting from PC
- Join the meeting from an Android device
- Set BackgroundBlur to video filter on the Android device
- Change the orientation of the Android device from portrait to landscape
- The height and width of the video should change on the basis of change of orientation

**Checklist:**

1. Have you successfully run `npm run build:release` locally? Yes

2. Do you add, modify, or delete public API definitions? If yes, has that been reviewed and approved? No

3. Do you change the wire protocol, e.g. the request method? If yes, has that been reviewed and approved? No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/backgroundblurprocessor/BackgroundBlurProcessorProvided.ts [hunk 1]
```diff
@@ -47,6 +47,20 @@ export default class BackgroundBlurProcessorProvided
     this.logger.info('BackgroundBlur processor successfully created');
     this.logger.info(`BackgroundBlur spec: ${this.stringify(this.spec)}`);
     this.logger.info(`BackgroundBlur options: ${this.stringify(options)}`);
+
+    screen.orientation.addEventListener('change', _event => {
```
Comment (hensmi-amazon) [inline src/backgroundblurprocessor/BackgroundBlurProcessorProvided.ts:51]: Can this be in `BackgroundFilterProcessor`?

File: src/backgroundblurprocessor/BackgroundBlurProcessorProvided.ts [hunk 2]
```diff
@@ -47,6 +47,20 @@ export default class BackgroundBlurProcessorProvided
...
+      if (
```
Comment (hensmi-amazon) [inline src/backgroundblurprocessor/BackgroundBlurProcessorProvided.ts:52]: Does this event listener need to be removed?

File: src/backgroundblurprocessor/BackgroundBlurProcessorProvided.ts [hunk 3]
```diff
@@ -47,6 +47,20 @@ export default class BackgroundBlurProcessorProvided
...
+        (screen.orientation.type.startsWith('portrait') &&
+          this.canvasCtx.canvas.height < this.canvasCtx.canvas.width) ||
```
Comment (hensmi-amazon) [inline src/backgroundblurprocessor/BackgroundBlurProcessorProvided.ts:54]: Can you put a comment that explains what this is doing in plain english? It's a bit tricky at first glance even if its doing something simple.

Comment (jhaarpan) [reply src/backgroundblurprocessor/BackgroundBlurProcessorProvided.ts:54]: Sure, I can add a comment

File: src/backgroundblurprocessor/BackgroundBlurProcessorProvided.ts (hunk 2 above)
Comment (jhaarpan) [reply src/backgroundblurprocessor/BackgroundBlurProcessorProvided.ts:52]: Yes, I will do that

File: src/backgroundblurprocessor/BackgroundBlurProcessorProvided.ts (hunk 1 above)
Comment (hensmi-amazon) [reply src/backgroundblurprocessor/BackgroundBlurProcessorProvided.ts:51]: What was the issue?

Comment (jhaarpan) [reply src/backgroundblurprocessor/BackgroundBlurProcessorProvided.ts:51]: It was not changing the orientation

Comment (hensmi-amazon) [reply src/backgroundblurprocessor/BackgroundBlurProcessorProvided.ts:52]: Is the listener not called? I don't see how it wouldn't work. At the very least can you put the listener function in shared code?

Comment (jhaarpan) [reply src/backgroundblurprocessor/BackgroundBlurProcessorProvided.ts:52]: The listener is called, but sometimes there is a delay which causes the canvas context to not change and hence causing the video to be garbled

File: src/backgroundfilter/BackgroundFilterProcessor.ts [hunk 4]
```diff
@@ -404,6 +404,7 @@ export default abstract class BackgroundFilterProcessor {
     this.scaledCanvas?.remove();
     this.scaledCanvas = undefined;
     this.logger.info(`${this.filterType} frame process destroyed`);
+    screen.orientation.onchange = null;
```
Comment (hensmi-amazon) [inline src/backgroundfilter/BackgroundFilterProcessor.ts:407]: You need to call `removeEventListener`

**Review by ltrung — APPROVED** [review]
